crypto/types/types.go

```diff
@@ -1,54 +1,36 @@
 package types
 
 import (
-	cmtcrypto "github.com/cometbft/cometbft/crypto"
+	cosmoscrypto "github.com/cosmos/crypto/types"
 	"github.com/cosmos/gogoproto/proto"
 )
 
+type (
+	// SdkPrivKey is a private key type parameterized by PubKey, which includes proto functions.
+	SdkPrivKey = cosmoscrypto.PrivKey[PubKey]
+	// Address is an alias of cosmoscrypto.Address
+	Address = cosmoscrypto.Address
+)
+
 // PubKey defines a public key and extends proto.Message.
 type PubKey interface {
 	proto.Message
-
-	Address() Address
-	Bytes() []byte
-	VerifySignature(msg, sig []byte) bool
-	Equals(PubKey) bool
-	Type() string
+	cosmoscrypto.PubKey
```

Comment:
i dont follow why these interfaces were moved? wont you have new apis?

Reply:
I've found a way to avoid having the Pubkey and Privkey ifaces copied across codebases. So in cosmos/crypto we have the "vanilla" ifaces, then in the sdk we can extend the ifaces to match the local requirements (mostly to be compatible with gogoproto). Same thing can be done is cometbft